### ISSUE TITLE
firewalld iptables generation broken in chef-client 13.7.16

### ISSUE THREAD

**Issue by dayne** [chef-cookbooks/firewall#185]
### Cookbook version 2.6.2 &  master 

Latest chef-client-13.7.16 has broken this cookbook's iptables configuration file generation on CentOS.

### Chef-client version
* Iptables configuration failing with `chef-client-13.7.16-1`
* works fine using `chef-client-13.6.4`

### Platform Details
Only the CentOS iptables generation is failing.  CentOS firewalld is fine.  Ubuntu and debian are also fine.

### Steps to Reproduce:

`kitchen converge centos` will result in:
* default-centos-6 failing
* default-centos-7 converges because it uses firewalld
* iptables-centos-7 failing

To verify this is a result of the 13.7.16 client update you can edit the `.kitchen.yml` and change `require_chef_omnibus: 13` to `require_chef_omnibus: 13.6.4`.

Make sure to do a `kitchen destroy` on failed converges as they have corrupt iptables config and a reconverge with older client wont' fix it.

### Expected Result:
All test systems should converge using latest chef-client.

### Actual Result:

The `iptables-centos-73` and `default-centos-69` on a converge now generates a bad iptables configuration using `chef-client-13.7.16-1`.  You can see full result (consistent across the centos iptables land) where it is only part of the file (missing the header that generates the INPUT filter and the COMMIT at end):

`[vagrant@iptables-centos-73 ~]$ sudo cat /etc/sysconfig/iptables`
```
# position 5
-A INPUT -p tcp -m tcp -m multiport --dports 7788 -m comment --comment "prepend" -j ACCEPT
# position 49
-A INPUT -s 192.168.99.99/32 -p tcp -m tcp -m comment --comment "block-192.168.99.99" -j REJECT
# position 50
-A INPUT -i lo -m comment --comment "allow loopback" -j ACCEPT
-A INPUT -p icmp -m comment --comment "allow icmp" -j ACCEPT
-A INPUT -p tcp -m tcp -m multiport --dports 22 -m comment --comment "allow world to ssh" -j ACCEPT
-A INPUT -p udp -m multiport --dports 60000:61000 -m comment --comment "allow world to mosh" -j ACCEPT
-A INPUT -m state --state RELATED,ESTABLISHED -m comment --comment "established" -j ACCEPT
-A INPUT -p tcp -m tcp -m multiport --dports 22 -m comment --comment "ssh22" -j ACCEPT
-A INPUT -p tcp -m tcp -m multiport --dports 2200,2222 -m comment --comment "ssh2222" -j ACCEPT
-A INPUT -p tcp -m tcp -m multiport --dports 1234 -m comment --comment "temp1" -j DROP
-A INPUT -p tcp -m tcp -m multiport --dports 1235 -m comment --comment "temp2" -j REJECT
-A INPUT -p tcp -m tcp -m multiport --dports 1236 -m comment --comment "addremove" -j ACCEPT
-A INPUT -p tcp -m tcp -m multiport --dports 1236 -m comment --comment "addremove2" -j DROP
-A INPUT -p 112 -m comment --comment "protocolnum" -j ACCEPT
-A INPUT -p tcp -m tcp -m multiport --dports 1111 -m comment --comment "same comment" -j ACCEPT
-A INPUT -p tcp -m tcp -m multiport --dports 5431,5432 -m comment --comment "same comment" -j ACCEPT
-A INPUT -p tcp -m tcp -m multiport --dports 1000:1100 -m comment --comment "range" -j ACCEPT
-A INPUT -p tcp -m tcp -m multiport --dports 1234,5000:5100,5678 -m comment --comment "array" -j ACCEPT
-A INPUT -p tcp -m tcp -m multiport --dports 5000:5100 -m comment --comment "RPC Port Range In" -j ACCEPT
-A OUTPUT -p tcp -m tcp -m multiport --dports 443,80 -m comment --comment "HTTP HTTPS" -j ACCEPT
```

**Comment by cstephen** [comment]
I encountered the same issue and switching to `require_chef_omnibus: 13.6.4` worked for me too. Thanks for tracking this down, @dayne!

**Comment by martinb3** [comment]
Looks related to:
https://github.com/chef/chef/issues/6784
https://github.com/chef/chef/issues/6792

**Comment by docwho76** [comment]
I too, have run into this issue, and had to temp fix it by pinning back to 13.6.4

**Comment by martinb3** [comment]
I've released a workaround in v2.6.3. Please give that a try and let us know if it's resolved.
